docs/plugin-write.md

```diff
@@ -0,0 +1,201 @@
+---
+id: plugin-write
+title: Writing Plugins
+---
+
+## Description
+
+Lets take a closer look at the minimal plugin instance we created in the [intro](/docs/plugin) part:
+
+```golang
+// Plugin is plugin instance
+type Plugin struct{}
+
+// Enable implements plugin.Plugin
+func (c *Plugin) Enable() error {
+	return nil
+}
+
+// Disable implements plugin.Plugin
+func (c *Plugin) Disable() error {
+	return nil
+}
+```
+
+Now it has only implemented the base [`Plugin`](https://godoc.org/github.com/gotify/plugin-api#Plugin) interface, and does nothing at all. To give it more functionalities, we need to implement more interfaces on `*Plugin`.
+
+## APIs
+
+### Intro
+
+The APIs are provided as interfaces and has generally two forms:
+
+The first form is called after plugins are started(after `Enabled` is called). They require the plugin to implements some methods which is called on user request to update information or provide feedbacks. An example is [`Displayer`](https://godoc.org/github.com/gotify/plugin-api#Displayer), which obtains an instruction generated by the plugin, which is then relayed to the user by gotify.
```

Comment:
This is not exactly correct, the display is shown even when the plugin is disabled. I guess it would be easier to understand to explicitly write to each plugin description when the methods are called. What do you think?

Reply:
Yep, I have added them in the comments.